### PR TITLE
feat: heatmaps TTL

### PR DESCRIPTION
**Pull request by pauldambra**
Let's not send an unbounded amount of heatmap data into the clickhouses

* allows setting TTL by day as well as by week
* adds a 90 day TTL to heatmaps table
* reduces the number of date filters available in the heatmap UI (for clickmaps they're already silly to offer)
* 
Screenshot 2024-04-24 at 19 31 09

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
Looks good I think

**Comment by pauldambra** [comment]
paging @fuziontech 

(I think we could do more here to reduce what we store but this is a nice stop-gap)

File: posthog/clickhouse/kafka_engine.py [hunk 1]
```diff
@@ -41,8 +42,8 @@ def kafka_engine(topic: str, kafka_host: str | None = None, group="group1") -> s
     return KAFKA_ENGINE.format(topic=topic, kafka_host=kafka_host, group=group, serialization="JSONEachRow")
 
 
-def ttl_period(field: str = "created_at", weeks: int = 3):
-    return "" if settings.TEST else f"TTL toDate({field}) + INTERVAL {weeks} WEEK"
+def ttl_period(field: str = "created_at", amount: int = 3, unit: Literal["DAY", "WEEK"] = "WEEK") -> str:
```
Comment (fuziontech) [inline posthog/clickhouse/kafka_engine.py:45]: Nice.

**Review by fuziontech — APPROVED** [review]
Less is more

**Pull request merged**